remotehttp_test.go

```diff
@@ -40,3 +43,259 @@ func TestHTTPStoreURL(t *testing.T) {
 		})
 	}
 }
+
+func TestHasChunk(t *testing.T) {
+	var attemptCount int
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		attemptCount++
+		switch r.URL.String() {
+		case "/0000/0000000100000000000000000000000000000000000000000000000000000000.cacnk":
+			w.WriteHeader(http.StatusOK)
+		case "/0000/0000000200000000000000000000000000000000000000000000000000000000.cacnk":
+			w.WriteHeader(http.StatusNotFound)
+		case "/0000/0000000300000000000000000000000000000000000000000000000000000000.cacnk":
+			w.WriteHeader(http.StatusBadRequest)
+		case "/0000/0000000400000000000000000000000000000000000000000000000000000000.cacnk":
+			w.WriteHeader(http.StatusForbidden)
+		case "/0000/0000000500000000000000000000000000000000000000000000000000000000.cacnk":
+			w.WriteHeader(http.StatusBadGateway)
+			io.WriteString(w, "Bad Gateway")
+		case "/0000/0000000600000000000000000000000000000000000000000000000000000000.cacnk":
+			if attemptCount >= 2 {
+				w.WriteHeader(http.StatusOK)
+			} else {
+				w.WriteHeader(http.StatusBadGateway)
+				io.WriteString(w, "Bad Gateway")
+			}
+		case "/0000/0000000700000000000000000000000000000000000000000000000000000000.cacnk":
+			if attemptCount >= 3 {
+				w.WriteHeader(http.StatusNotFound)
+			} else {
+				w.WriteHeader(http.StatusBadGateway)
+				io.WriteString(w, "Bad Gateway")
+			}
+		default:
+			w.WriteHeader(http.StatusBadRequest)
+		}
+	}))
+	defer ts.Close()
+	u, _ := url.Parse(ts.URL)
+
+	tests := map[string]struct {
+		chunkId      ChunkID
+		hasChunk     bool
+		hasError     bool
+		attemptCount int
+	}{
+		// The default case is a successful chunk test operation
+		"chunk exists": {ChunkID{0, 0, 0, 1}, true, false, 1},
+		// HTTP 404 Not Found - Testing a chunk that does not exist should result in an immediate 'does not exist' response
+		"chunk does not exist": {ChunkID{0, 0, 0, 2}, false, false, 1},
+		// HTTP 400 Bad Request - should fail immediately
+		"bad request": {ChunkID{0, 0, 0, 3}, false, true, 1},
+		// HTTP 403 Forbidden - should fail immediately
+		"forbidden": {ChunkID{0, 0, 0, 4}, false, true, 1},
+		// HTTP 503 Bad Gateway - should retry, but ultimately fail
+		"permanent 503": {ChunkID{0, 0, 0, 5}, false, true, 5},
+		// HTTP 503 Bad Gateway - should retry, and a subsequent successful call should return that the chunk exists
+		"temporary 503, then chunk exists": {ChunkID{0, 0, 0, 6}, true, false, 2},
+		// HTTP 503 Bad Gateway - should retry, and a subsequent successful call should report that the chunk does not exist immediately
+		"temporary 503, then chunk does not exist": {ChunkID{0, 0, 0, 7}, false, false, 3},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			u.Path = "/"
+			s, err := NewRemoteHTTPStore(u, StoreOptions{ErrorRetry: 5, ErrorRetryBaseInterval: time.Duration(1) * time.Microsecond})
```

Comment:
Don't really need the `time.Duration(1)` bit here, just `time.Microsecond` will be enough

Reply:
Thanks, will fix.